src/compiler/scala/tools/nsc/transform/patmat/MatchAnalysis.scala

```diff
@@ -372,10 +388,11 @@ trait MatchApproximation extends TreeAndTypeAnalysis with ScalaLogic with MatchT
       }
 
       // special-case: interpret pattern `List()` as `Nil`
+      // as of 2.13, List.unapply returns an UnapplySeqWrapper (rather than a List)
       // TODO: make it more general List(1, 2) => 1 :: 2 :: Nil  -- not sure this is a good idea...
       private val rewriteListPattern: PartialFunction[TreeMaker, Prop] = {
         case p @ ExtractorTreeMaker(_, _, testedBinder)
-          if testedBinder.tpe.typeSymbol == ListClass && p.checkedLength == Some(0) =>
+          if testedBinder.tpe.typeSymbol == UnapplySeqWrapperClass && p.checkedLength == Some(0) =>
```

Comment:
Doesn't this catch other uses of `SeqFactory.unapplySeq`, like 	`case Seq()`?

Reply:
Yeah.  Which is regrettable, but I couldn't convince myself that that was bad so I went with it...  Do you think it'll affect other things and/or have a better idea to fixing the regression?